frontend/actions/siteActions.js

```diff
@@ -31,6 +33,16 @@ export default {
       .catch(alertError);
   },
 
+  addUserToSite({ owner, repository }) {
+    return federalist.addUserToSite({ owner, repository })
+      .then(dispatchUserAddedToSiteAction)
+      .then(updateRouterToSitesUri)
+      // rather than display an alert error for this action
+      // we'll instead want to show the additional fields necessary
+      // for adding a completely new site to Federalist
+      .catch(dispatchShowAddNewSiteFieldsAction);
```

Comment:
We may want to look out for bugs here. Could we check the error code to make sure it's a 404 or something like that? I don't think we want to swallow other bugs, like "you don't have read access" or something like that.

Reply:
That's a good idea. The way it is currently written it won't get a read access bug, but there maybe could be a case where something else server side throws an exception (like some bad code that triggers a `500` or something).